### DISCUSSION TITLE
Easy Trading V3

### DISCUSSION THREAD

**Discussion by Life-Is-Great** [sunflower-land/sunflower-land#1230] [General]
Hello all.

V3 is up as a main update point. (Next planned update will bring statistics to you)
(Don't forget to give the site permission for audio if you want to hear your "Listing Notification")

Link:
V3: [http://sunflower.alwaysdata.net/](http://sunflower.alwaysdata.net/) (Old link)
V4: [https://sfl.land](https://sfl.land) (In progress & coming soon)

![image](https://user-images.githubusercontent.com/108883042/180439586-bf72b53e-44b7-418c-9361-dc2f6c919b75.png)

What is done in V3:
* Resource info update without refresh and in *realtime (every 5 seconds)
(Before, site was updated when user was closing down resource popup)

* Resource Card ->
   - Added listings total indicator
   - Added lowest price 1xresource indicator

* Listings Table
  - Added copied to clipboard indicator

* Settings
  - Click sound (On / Off) setting added
  - Added "Pricing currency change"
     (possibility to view prices in SFL, MATIC and USD)
  - Listing Notification
     --> Chose between Off, Sound+Popup & Only Popup ) 

If you like me to do changes. Hit me up on discord
(@GodIsGreat)

If you would like to buy me a coffee and keep me warm while I work to make gameplay better for the community you can donate to this wallet:
**0x6DebC88D809e9c6Adf6ca60Aa326595d14C39181**

**Comment by adamhannigan** [comment]
Love to see these community tools popping up! Such a great contribution to the community ❤️ 

I will add the message I sent in Discord just in case other developers view this message.

The Trader.sol protocol is designed in a way that gives lots of public APIs and capabilities for developers to read data and events.

https://polygonscan.com/address/0x1056A4ea959e3d84D6DC56Ac06B8Ff5B84648400#events

Using web3 `getPastEvents` you can query for items being listed and also filter them by item type and listed farm. These ABIs are available in the `Trader.json` file in the repo.

Excited to see what can be built! 💪

**Reply by Life-Is-Great** [reply]
Best m8!
Thank you for your gift and check out the updated version ;) =D

Hope that you like it.

**Comment by Q4FOIL** [comment]
@Life-Is-Great If I am allowed, I would like to be able to contribute graphically to the project making it more uniform to SFL

**Reply by Life-Is-Great** [reply]
Hey m8. Absolutely. Hit me up on Discord

**Reply by Q4FOIL** [reply]
I need your number -> #

@Life-Is-Great

**Reply by Life-Is-Great** [reply]
oh. I thot that you had it up there. But here m8: GodIsGreat#1718

**Comment by Q4FOIL** [comment]
@Life-Is-Great is suggest to merge this idea with this 

https://github.com/sunflower-land/sunflower-land/discussions/1234


What's need :

- "Request Area" when user ask on public to find X resources to X SFL.
|- "Farm ID Request"
|- "Resource"
|- "Quantity"
|- "How much you want to spend?"
|| **To agree, the seller will only have to respond by commenting on that request with his farm ID** ||

Need a place where people can ask resource and maybe don't want to spend more to buy an huge quantity 

In this way we create a total decentralized market.

**Reply by Life-Is-Great** [reply]
Hello m8.

I have so far added almost all requests asked from community.
Will definitely consider this one to.
Thought. This does not have that much benefit because you can just ask in discord for someone to add exact amount of resources and then monitor or add a listing notification that matches that trade criteria.
![image](https://user-images.githubusercontent.com/108883042/180996118-e299626d-c53a-4237-b9f5-eaa0e2445152.png)

**Reply by Q4FOIL** [reply]
> Hello m8.
> 
> I have so far added almost all requests asked from community. Will definitely consider this one to. Thought. This does not have that much benefit because you can just ask in discord for someone to add exact amount of resources and then monitor or add a listing notification that matches that trade criteria. ![image](https://user-images.githubusercontent.com/108883042/180996118-e299626d-c53a-4237-b9f5-eaa0e2445152.png)

You can make this more simply :/

Anyway great job!

❤️

**Reply by Life-Is-Great** [reply]
Happy to hear what you have on your mind when you say more simple?
<3

**Comment by Q4FOIL** [comment]
https://sunflower-land.com/play/#/goblins?viewListingsForLand=IDFARM <- CREATE A BUTTON



Add this trick of every listing

❤️

**Reply by Life-Is-Great** [reply]
Will do in v4 m8. (And much more =D )
Happy to hear what your wishes <3
